pkg/sql/pg_catalog.go

```diff
@@ -1436,8 +1436,43 @@ CREATE TABLE pg_catalog.pg_enum (
   enumsortorder FLOAT4,
   enumlabel STRING
 )`,
-	populate: func(_ context.Context, p *planner, _ *DatabaseDescriptor, addRow func(...tree.Datum) error) error {
-		// Enum types are not currently supported.
+	populate: func(ctx context.Context, p *planner, dbContext *DatabaseDescriptor, addRow func(...tree.Datum) error) error {
+		h := makeOidHasher()
+		descs, err := p.Tables().getAllDescriptors(ctx, p.txn)
+		if err != nil {
+			return err
+		}
+		for _, desc := range descs {
+			typDesc, ok := desc.(*sqlbase.TypeDescriptor)
+			if !ok {
+				continue
+			}
+			if dbContext != nil && typDesc.ParentID != dbContext.ID {
+				continue
+			}
+			if typDesc.Kind != sqlbase.TypeDescriptor_ENUM {
+				continue
+			}
+			// Generate a row for each member of the enum. We don't represent enums
+			// internally using floats for ordering like Postgres, so just pick a
+			// float entry for the rows.
+			typ := types.MakeEnum(uint32(typDesc.ID))
+			if err := typDesc.HydrateTypeInfo(typ); err != nil {
```

Comment:
stay hydrated!
(why didn't we call this `InitTypeInfo`? hah.

Reply:
IDK, someone said Hydrate once and then I couldn't get it out of my head